### PR TITLE
[AMBARI-26135] Ambari-Metrics: Upgrade net.sf.ehcache to 3.10.0 to resolve CVE

### PR DESCRIPTION
**Pull request by sandeep318kumar**
## What changes were proposed in this pull request?
upgrading ehcache from 2.10.0 to 3.10.0

(Please fill in changes proposed in this fix)

## How was this patch tested?
Tested by creating a local cluster with this patch.

(Please explain how this patch was tested. Ex: unit tests, manual tests)
(If this patch involves UI changes, please attach a screen-shot; otherwise, remove this)

Please review [Ambari Contributing Guide](https://cwiki.apache.org/confluence/display/AMBARI/How+to+Contribute) before opening a pull request.

### CODE REVIEW

**Comment by sandeep318kumar** [comment]
@JiaLiangC @arshadmohammad @virajjasani can you help in review?

ehcache upgrade PR already merged in ambari. Ref: https://github.com/apache/ambari/pull/3825

File: ambari-metrics-timelineservice/src/main/java/org/apache/ambari/metrics/core/timeline/source/cache/InternalMetricsCache.java [hunk 1]
```diff
@@ -181,49 +161,14 @@ public void putAll(Collection<TimelineMetrics> metrics) {
...
-      return null;
-    }
-
-    @Override
-    public void dispose() {
-      // do nothing
+    public void onEvent(CacheEvent<? extends InternalMetricCacheKey, ? extends InternalMetricCacheValue> event) {
+      if (event.getType() == EventType.EVICTED) {
+        InternalMetricCacheKey key = event.getKey();
+        LOG.warn("Evicting element from internal metrics cache, metric => " + key
```
Comment (virajjasani) [inline ambari-metrics-timelineservice/src/main/java/org/apache/ambari/metrics/core/timeline/source/cache/InternalMetricsCache.java:169]: nit: can we use {} placeholders?

**Review by virajjasani — COMMENTED** [review]
One nit, +1 overall

File: ambari-metrics-timelineservice/src/main/java/org/apache/ambari/metrics/core/timeline/source/cache/InternalMetricsCache.java (hunk 1 above)
Comment (sandeep318kumar) [reply ambari-metrics-timelineservice/src/main/java/org/apache/ambari/metrics/core/timeline/source/cache/InternalMetricsCache.java:169]: @virajjasani Apache Commons Logging does not support {} placeholders directly.

**Review by arshadmohammad — APPROVED** [review]
LGTM +1

**Pull request merged**